### PR TITLE
[M] ENT-2215: truncate correlation id to 40 characters

### PR DESCRIPTION
**Pull request by abhiskum**
- Truncate correlation id to 40 characters instead of dropping it.

### CODE REVIEW

File: common/src/main/java/org/candlepin/common/filter/LoggingFilter.java [hunk 1]
```diff
@@ -46,8 +46,7 @@
     private static Logger log = LoggerFactory.getLogger(LoggingFilter.class);
 
     public static final String CSID_KEY = "csid";
-    private static final int CSID_MAX_LENGTH = 40;
```
Comment (wottop) [inline common/src/main/java/org/candlepin/common/filter/LoggingFilter.java]: Please keep and use the constant.

File: common/src/main/java/org/candlepin/common/filter/LoggingFilter.java [hunk 2]
```diff
@@ -95,11 +94,11 @@ public void doFilter(ServletRequest request, ServletResponse response,
 
             if (correlationId != null) {
                 if (CSID_REGEX.matcher(correlationId).matches()) {
-                    MDC.put(CSID_KEY, correlationId);
+                    MDC.put(CSID_KEY,
```
Comment (Ceiu) [inline common/src/main/java/org/candlepin/common/filter/LoggingFilter.java]: I really don't feel truncating is the right approach here, but if that's what we're going to do, we should still output a log message indicating that we've truncated the original value down, and what that value was so we can at least track and figure out what happened in the event that it happens or when there's a collision caused by it.

File: common/src/main/java/org/candlepin/common/filter/LoggingFilter.java (hunk 1 above)
Comment (abhiskum) [reply common/src/main/java/org/candlepin/common/filter/LoggingFilter.java:49]: Fixed.

File: common/src/main/java/org/candlepin/common/filter/LoggingFilter.java (hunk 2 above)
Comment (abhiskum) [reply common/src/main/java/org/candlepin/common/filter/LoggingFilter.java]: Fixed.

**Comment by wottop** [comment]
ack

**Pull request merged**